ops/terraform/services/server/modules/bfd_server_asg/main.tf

```diff
@@ -230,65 +259,71 @@ resource "aws_cloudwatch_metric_alarm" "filtered_networkin_low" {
       }
       metric_name = "NetworkOut"
       namespace   = "AWS/EC2"
-      period      = 60
+      period      = local.scaling_alarms_config.scale_in.eval_period
+      stat        = "Average"
+    }
+  }
+
+  metric_query {
+    id          = "m3"
+    return_data = false
+
+    metric {
+      dimensions = {
+        AutoScalingGroupName = aws_autoscaling_group.main.name
+      }
+      metric_name = "GroupDesiredCapacity"
+      namespace   = "AWS/AutoScaling"
+      period      = local.scaling_alarms_config.scale_in.eval_period
       stat        = "Average"
     }
   }
 
   metric_query {
     expression  = "IF(m2/m1 > 0.01, m1, 0)"
-    id          = "e1"
+    id          = "networkin"
     label       = "FilteredNetworkIn"
+    return_data = false
+  }
+
+  metric_query {
+    id    = "e2"
+    label = "Set to ${each.value.desired_capacity} capacity units"
+    expression = "IF(${join(" && ", compact([
+      each.value.begin > 0 ? "networkin > ${each.value.begin}" : null,
+      each.value.end != null ? "networkin <= ${each.value.end}" : null,
+      "m3 > ${each.value.desired_capacity}"
+    ]))}, 1)"
```

Comment:
Dumb question but should this be `1 * ${each.value.desired_capacity}`..?

Reply:
Assuming you're asking about line 296: this `1` could really be any value, assuming the value broke the alarm's threshold. The value returned by the `IF()`, in this case `1`, can be thought of a boolean value indicating whether the ASG _could_ be scaled-down at that given period. If there are 15 consecutive periods of this boolean value being "true" (`1`), then the alarm enters `ALARM` and the scale-in policy can evaluate.

Does that make sense? I can see how the label for this expression is a little vague, so I can definitely reword it to make the above more clear. Definitely not a dumb question, either -- AWS's scaling mechanisms aren't really designed for this particular scaling configuration so this is coloring outside the lines a bit.

Reply:
Gotcha! Thank you for elaborating - makes total sense.